assertion/function/assertiontree/assertion_node.go

```diff
@@ -106,3 +109,7 @@ func (n *assertionNodeCommon) Size() int {
 	}
 	return size
 }
+
+func (n *assertionNodeCommon) BuildExpr() ast.Expr {
+	return n.originalExpr
+}
```

Comment:
I do appreciate the encapsulations here, but I wonder if it's really necessary anymore since we no longer need polymorphisms here and I don't see such need for the near future.

We can just make `originalExpr` exported and use the field instead, which is a lot simpler. (we can always encapsulate later when there is a need)